### PR TITLE
Function for Download Button

### LINKED ISSUE #381: Export Truth Tables
I'm using CircuitVerse for lots uni stuff and truth tables would be useful for testing if my circuit behaves as expected, minimising etc..

When editing the circuit (where the whole thing is composed into one node and I can set the width and height) there could be an option to pop out the truth table. Additionally you should be able to have it rendered into the export image / SVG aswell

**Are you working on this? (Yes/No)**
No

### PR DESCRIPTION
**Pull request by Shivansh2407**
Fixes #381 #514 

#### Describe the changes you have made in this pr -
I have added a download button to the Boolean Logic Table Div , So when User clicks on the Download Button the Table will be saved as a PDF .

### Screenshots of the changes (If any) - 
patch-15

### CODE REVIEW

**Comment by Shivansh2407** [comment]
@tachyons Please review this and let me know how can I fix the codeclimate issues here.

**Comment by Shivansh2407** [comment]
Fixed the Code Climate Issues . 👍 Let me know your views about the Download Button @satu0king and @tachyons Sir

**Comment by Shivansh2407** [comment]
@satu0king and @tachyons  what is your opinion on this?

**Comment by eKuG** [comment]
Not bad! @tachyons, have a look.

**Comment by Shivansh2407** [comment]
@satu0king , @tachyons Please review this 👍

File: public/js/combinationalAnalysis.js [hunk 1]
```diff
@@ -101,8 +101,24 @@ function createBooleanPrompt(inputListNames,outputListNames,scope=globalScope){
             // //console.log(dataSample);
             drawCombinationalAnalysis(minmizedCircuit,inputListNames,outputListNames,scope)
         },
-
-        }
+      },
+      {
+        text: "Download",
+        click: function() {
+           var sTable = document.getElementById('combinationalAnalysis').innerHTML;
```
Comment (ciphereck) [inline public/js/combinationalAnalysis.js]: can we change this variable to booleanLogicTable? I think this way it's better to understand what this variable contains
Can we make this as constant? const booleanLogictable=....

File: public/js/combinationalAnalysis.js [hunk 2]
```diff
@@ -101,8 +101,24 @@ function createBooleanPrompt(inputListNames,outputListNames,scope=globalScope){
...
+           var style = "<style> table {font: 20px Calibri;} table, th, td {border: solid 1px #DDD;} padding: 2px 3px;text-align: center;} </style>";
+           var win = window.open('', '', 'height=700,width=700');
```
Comment (ciphereck) [inline public/js/combinationalAnalysis.js]: I think this is a new browserTab. Maybe you can change the variable to **browserTab**

File: public/js/combinationalAnalysis.js [hunk 3]
```diff
@@ -101,8 +101,24 @@ function createBooleanPrompt(inputListNames,outputListNames,scope=globalScope){
...
             drawCombinationalAnalysis(minmizedCircuit,inputListNames,outputListNames,scope)
         },
-
-        }
+      },
+      {
+        text: "Download",
+        click: function() {
+           var sTable = document.getElementById('combinationalAnalysis').innerHTML;
+           var style = "<style> table {font: 20px Calibri;} table, th, td {border: solid 1px #DDD;} padding: 2px 3px;text-align: center;} </style>";
```
Comment (ciphereck) [inline public/js/combinationalAnalysis.js]: I think it's better to make this variable final or const

File: public/js/combinationalAnalysis.js [hunk 4]
```diff
@@ -101,8 +101,24 @@ function createBooleanPrompt(inputListNames,outputListNames,scope=globalScope){
             // //console.log(dataSample);
             drawCombinationalAnalysis(minmizedCircuit,inputListNames,outputListNames,scope)
         },
-
-        }
+      },
+      {
+        text: "Download",
```
[the remark this answers is not in the document]
Comment (satu0king) [reply public/js/combinationalAnalysis.js]: I agree with @ciphereck

**Comment by satu0king** [comment]
Good Job, The aesthetics can be improved. I would like to see copy table to clipboard option and an option to paste from clipboard as well. @Shivansh2407

**Comment by satu0king** [comment]
I have merge this PR by pushing into master branch directly (had to make some changed)

**Comment by Shivansh2407** [comment]
Thank you @satu0king. Should I implement the Copy and Paste feature in a separate PR then ? Also I will make the look and feel more interactive and better 👍🏻

**Pull request closed without merging**